pkgs/development/libraries/qt-6/default.nix

```diff
@@ -174,16 +181,28 @@ let
         } ./hooks/qmake-hook.sh;
     };
 
-  # TODO(@Artturin): convert to makeScopeWithSplicing'
-  # simple example of how to do that in 5568a4d25ca406809530420996d57e0876ca1a01
-  baseScope = lib.makeScope newScope addPackages;
 
-  bootstrapScope = baseScope.overrideScope'(final: prev: {
+  baseScope = lib.makeScopeWithSplicing' {
+    inherit splicePackages newScope;
+  } {
+    otherSplices = {
```

Comment:
Check how qt5 does it and omit the unnecessary args

Reply:
34d8710  (squashed)

Reply:
Remove the unused args and use the prefilled makeScopeWithSplicing' from pkgs also remove the keep and extra

Reply:
> also remove the keep and extra

I wish I understood what those did.

Reply:
efdddc051f64

Reply:
> > also remove the keep and extra
> 
> I wish I understood what those did.

https://github.com/NixOS/nixpkgs/pull/265045